sodym/named_dim_arrays.py

```diff
@@ -225,6 +226,29 @@ def to_df(self):
         df["value"] = self.values.flatten()
         return df
 
+    def split(self, dim_letter) -> dict:
+        """Reverse the named_dim_array_stack, returns a dictionary of `NamedDimArray`s
+        associated with the item in the dimension that has been split.
+        Method can be applied to `NamedDimArray`s, `StockArray`s, `Parameter`s and `Flow`s.
+        """
+        if dim_letter not in self.dims.letters:
+            raise ValueError('Dimension to split on must exist in the named_dim_array')
+        smaller_dimensions = self.dims.drop(dim_letter, inplace=False)
+        extracted_dimension = self.dims[dim_letter]
+        reorganised_values = rearrange(
+            self.values, f'{self.dims.spaced_string} -> {dim_letter} {smaller_dimensions.spaced_string}'
+        )
+        kwargs = {}
+        if isinstance(self, Flow):
+            kwargs = {"from_process": self.from_process, "to_process": self.to_process}
```

Comment:
See my other comment: Would be nice to make the additional `__fields__` kwargs more general.

Reply:
yes, agree